logstash-core/src/main/java/org/logstash/ackedqueue/io/FileCheckpointIO.java

```diff
@@ -64,10 +63,12 @@ public Checkpoint write(String fileName, int pageNum, int firstUnackedPageNum, l
 
     @Override
     public void write(String fileName, Checkpoint checkpoint) throws IOException {
-        Path path = Paths.get(dirPath, fileName);
-        final byte[] buffer = new byte[BUFFER_SIZE];
         write(checkpoint, buffer);
-        Files.write(path, buffer, WRITE_OPTIONS);
+        buffer.flip();
+        try (FileOutputStream out = new FileOutputStream(Paths.get(dirPath, fileName).toFile())) {
```

Comment:
@jordansissel I like putting a comment here explaining the use of a direct buffer because of I/O (and actually also the CRC32 call ... this is a lot less obvious to only be 0GC using a direct buffer imo :)).
I don't really like `assert` tbh, it's such a standard spot in Java 8 (Direct way caught up to Heap performance wise, even for serializing tiny old `int` the direct one is ahead on OSX for me). 
Seems like this invites a bunch of weird (in the sense that I have never seen anyone asserting `isDirect() == true` anywhere) `assert` all over the place. I think this is really something that should be looked out for in code reviews instead (which would be made easy by having the comment I guess :D).